### ISSUE TITLE
DeepAR native implementation

### ISSUE THREAD

**Issue by alex-hse-repository** [etna-team/etna#85]
### 🚀 Feature Request

Add native implementation of DeepAR model in the library

### Proposal

1. Read the [paper](https://arxiv.org/pdf/1704.04110.pdf)
2. Implement `DeepARModel`, see [RNNModel ](https://github.com/etna-team/etna/blob/master/etna/models/nn/rnn.py#L32) as an example
  - `DeepARNet(DeepBaseNet)` -- implement the network itself as [LightningModule](https://lightning.ai/docs/pytorch/stable/common/lightning_module.html)
  - `DeepARModel(DeepBaseModel)` -- implement etna model wrapping the network, implement`params_to_tune` 
  - Referens implementations: [PytorchForecasting](https://pytorch-forecasting.readthedocs.io/en/stable/api/pytorch_forecasting.models.deepar.DeepAR.html), [GluonTS](https://github.com/awslabs/gluonts/tree/dev/src/gluonts/torch/model/deepar), [Nixtla](https://github.com/Nixtla/neuralforecast/blob/main/neuralforecast/models/deepar.py) + implementations from [paperswithcode](https://paperswithcode.com/paper/deepar-probabilistic-forecasting-with)
3.  Try to reproduce metrics from the paper, see similar experiments for [DeepState](https://gitlab.tcsbank.ru/etna/experiments/-/tree/45480145a6dbef9f80b215b14765233fe88279b2/experiments/ETNA-1022-v2), [PatchTS](https://gitlab.tcsbank.ru/etna/experiments/-/merge_requests/321)
- Find datasets from the paper(see etna-datasets)
- Run experiment with configuration from the paper
- Run the same experiment with implementation from PytorchForecasting
- Add metric from your experiments and from the paper into PR 
4. Update example [notebook](https://github.com/etna-team/etna/blob/master/examples/202-NN_examples.ipynb)

### Test cases

- Test `params_to_tune` similar to the other models
- Add sanity checks similar to the other models

### Additional context

We want to replace current implementation from PytorchForecasting without bumping the major version, we should somehow warn users that new implementation might work slightly different

**Comment by egoriyaa** [comment]
Add:
1) NormalDistributionLoss
2) NegativeBinomialDistributionLoss
3) Parameters of mean and std of distribution
4) Count loss for both train context and predictions
5) Samples with context starts before ts begins(pad it)
6) Weighted sampling
7) Scale targets and parameters of distribution
